### ISSUE TITLE
[Bug] Resume with more than one sheet can't be print beyond 1st page

### ISSUE THREAD

**Issue by kartiksingla** [AmruthPillai/Reactive-Resume#1834]
### Is there an existing issue for this?

- [X] Yes, I have searched the existing issues and none of them match my problem.

### Product Variant

Cloud (http://rxresu.me)

### Current Behavior

Resume is longer than 1 A4 page. After exporting it to PDF, when tried to print it, unable to print anything after 1st page _(after page break)_. 
**Problem:** Exports the PDF with one long single page.

### Expected Behavior

Should split the pages by line splitter into separate page when exported in PDF along with some margin on the top

### Steps To Reproduce

1. Create a resume with content flowing to 2nd page
2. From the right side, enable Show Break Line
3. Export PDF
4. Print by selecting A4 or Letter or A3, you will see the content **only** before the page break on the website is coming in print view.

### What template are you using?

Onyx

**Comment by randomsynergy** [comment]
Experiencing the same issue.  The PDF is one LONG page with no pagiation. FOllowing the instructions in the guide: 
https://docs.rxresu.me/product-guides/creating-a-multi-page-resume

seems you can span the experience across multiple pages but when trying to add experience to the second page it removes it from the first, unlike the animation / details in the guide above.

**Comment by AmruthPillai** [comment]
> Experiencing the same issue with Self Hosted.  
> 
> The PDF is one LONG page with no pagination. FOllowing the instructions in the guide: 
> 
> https://docs.rxresu.me/product-guides/creating-a-multi-page-resume.
> 
> seems you can span the experience across multiple pages but when trying to add experience to the second page it removes it from the first, unlike the animation / details in the guide above. Would be great if this feature worked as show in the guide, properly breaking sections that space across multiple pages and implement page numbering as expected

The animation of the docs is a screen recording of the actual app, so it should work as expected. I think what is important which has been missed in the docs is that you'll need to break up your Experience section into two separate sections, which is done by basically creating a new custom section and also calling it "Experience". Now you can move the new section to a new page while the old one remains on the first page.

I understand that this is not an ideal process, I would love for this to be automated too, to push sections into their own page intelligently without abrupt breaks within the text. Unfortunately I'm just not that smart 😅

**Comment by radiatewishbone** [comment]
@AmruthPillai the current system is an imperfect solution for two reasons:

1. Attempting to "split" the PDF by guessing where a good place would be to add a second work experience section means that you will never have a perfect A4 split, because it's impossible to guess the exact place to split the page to create an A4 page.
2. The documentation says that there is "no need" for A4 pages on a PDF anymore, because nobody prints them out. This is untrue. In my industry, people _do_ still print resumes. Also, for many job applications I make online, application forms require you to upload PDFs, not provide a link to a PDF. Hiring teams will then want to print the PDFs out themselves to read.

It would be very good if you could add the option to split the PDF into a specified page size.

**Comment by AmruthPillai** [comment]
> @AmruthPillai the current system is an imperfect solution for two reasons:
> 
> 1. Attempting to "split" the PDF by guessing where a good place would be to add a second work experience section means that you will never have a perfect A4 split, because it's impossible to guess the exact place to split the page to create an A4 page.
> 2. The documentation says that there is "no need" for A4 pages on a PDF anymore, because nobody prints them out. This is untrue. In my industry, people _do_ still print resumes. Also, for many job applications I make online, application forms require you to upload PDFs, not provide a link to a PDF. Hiring teams will then want to print the PDFs out themselves to read.
> 
> It would be very good if you could add the option to split the PDF into a specified page size.

There's a toggle/checkbox in the right sidebar of the app, while building the resume, called Toggle Break Line, which shows you a line where the page ends. All you would have to do then is move the sections that cross the line into their own page.

**Comment by radiatewishbone** [comment]
> > @AmruthPillai the current system is an imperfect solution for two reasons:
> > 
> > 1. Attempting to "split" the PDF by guessing where a good place would be to add a second work experience section means that you will never have a perfect A4 split, because it's impossible to guess the exact place to split the page to create an A4 page.
> > 2. The documentation says that there is "no need" for A4 pages on a PDF anymore, because nobody prints them out. This is untrue. In my industry, people _do_ still print resumes. Also, for many job applications I make online, application forms require you to upload PDFs, not provide a link to a PDF. Hiring teams will then want to print the PDFs out themselves to read.
> > 
> > It would be very good if you could add the option to split the PDF into a specified page size.
> 
> There's a toggle/checkbox in the right sidebar of the app, while building the resume, called Toggle Break Line, which shows you a line where the page ends. All you would have to do then is move the sections that cross the line into their own page.

And just to clarify, are these break lines set to A4 by default? How would I use this to create a PDF with multiple separate A4 pages?

Another issue with this system:

Let's say I have lots of work experience that would go beyond the break line. With the current system, I need to create a new work experience section, and place this second work experience section after the break line.

The problem with this is that it means "Work Experience" will become a title twice in the same document. It will say "Work Experience" on the first page part of the first section, and then after the break line it will say "Work Experience" again, for the second work experience section. I don't want my CV to have the same title twice.

**Comment by amizzo87** [comment]
+1 to this. Love this app but there needs to be proper pagination/spillover; the suggested workaround is suboptimal at best.

**Comment by aboeing** [comment]
+1. 
I cant get the workaround to work properly. I am not sure which industry everyone else works in, but in all the industries I've ever worked in the resume has always been printed out.

The documentation here: https://docs.rxresu.me/product-guides/creating-a-multi-page-resume should be updated, as the animated gif is extremely misleading.

Several challenges: 
1. Formatting: I am unable to insert 'whitespace' to properly pad sections, and you end up with multiple copies of the same section.
2. Useability: It took me a long time to figure out how to add a custom section, then recreate the content, and figure out how to delete old content
3. PDF: I have not been able to find a PDF tool that will let me 'cut' or 'crop' a page in order to generate multiple A4 pages.

**Comment by atsaloli** [comment]
See also https://superuser.com/questions/1836244/how-to-split-a-long-single-page-pdf-in-multiple-pages

**Comment by kevinawoo** [comment]
I'm still experiencing the issue described, but found a workaround.

- Open your resume via public link
- Open the Print Dialog and adjust some settings
    <kbd>![CleanShot 2024-07-09 at 22 49 57](https://github.com/AmruthPillai/Reactive-Resume/assets/3469532/131613b8-2ee1-4191-b201-28cb22905493)</kbd>
  - Margins: `None`
  - select `Background Graphics`

- Save it as a PDF version



It's still kind of buggy between page breaks, the text will be split between pages, here's an example:

<kbd>![CleanShot 2024-07-09 at 22 54 26](https://github.com/AmruthPillai/Reactive-Resume/assets/3469532/f448e7e8-eed8-4a16-a1ce-e691a403aa0c)
</kbd>


Also don't click the button download, that'll generate the 1 pager.

**Comment by aboeing** [comment]
Thanks for sharing your method - this works! - which is great; but as you said it doesn't make nice page splits; it also joins all the pre-defined pages into one long webpage.

It would be better to be able to set page breaks within the tool, I hope Reactive Resume will support such a feature in future.

If anyone else isn't sure where to find it, the public link can be enabled via the sharing section on the right side of the screen.

**Comment by AMDphreak** [comment]
Without having read everyone else's comments (only a bit here and there), there is a way to insert new pages. It's strange and unintuitive.

![image](https://github.com/user-attachments/assets/62be35f0-a4af-45de-84fa-d0b9109a06f1)

**Comment by radiatewishbone** [comment]
> > > @AmruthPillai the current system is an imperfect solution for two reasons:
> > > 
> > > 1. Attempting to "split" the PDF by guessing where a good place would be to add a second work experience section means that you will never have a perfect A4 split, because it's impossible to guess the exact place to split the page to create an A4 page.
> > > 2. The documentation says that there is "no need" for A4 pages on a PDF anymore, because nobody prints them out. This is untrue. In my industry, people _do_ still print resumes. Also, for many job applications I make online, application forms require you to upload PDFs, not provide a link to a PDF. Hiring teams will then want to print the PDFs out themselves to read.
> > > 
> > > It would be very good if you could add the option to split the PDF into a specified page size.
> > 
> > 
> > There's a toggle/checkbox in the right sidebar of the app, while building the resume, called Toggle Break Line, which shows you a line where the page ends. All you would have to do then is move the sections that cross the line into their own page.
> 
> And just to clarify, are these break lines set to A4 by default? How would I use this to create a PDF with multiple separate A4 pages?
> 
> Another issue with this system:
> 
> Let's say I have lots of work experience that would go beyond the break line. With the current system, I need to create a new work experience section, and place this second work experience section after the break line.
> 
> The problem with this is that it means "Work Experience" will become a title twice in the same document. It will say "Work Experience" on the first page part of the first section, and then after the break line it will say "Work Experience" again, for the second work experience section. I don't want my CV to have the same title twice.


The fact is that all that has been done to address these complaints (that I and many others have) have been sub-par workarounds. Until these exact issues are addressed it's not a usable tool for me.

**Comment by ernietedeschi** [comment]
What would help immensely is the ability to 1) move elements of sections _between sections_, and/or 2) duplicate entire sections. Here's how: after putting in all of my experience and orienting my resume, I find that a single "Experience" section is simply too long for a single page no matter what. If I had the ability to either move elements across sections and/or duplicate sections, I could create an "Experience, cont." section on Page 2, and either move the spillover elements from page 1 into that one, or have that section start out as a duplicate of the Page 1 "Experience" section from which I then delete the repetitive elements. Either way, I could make a single section first to see where the breaks are, and then I wouldn't have to input everything all over again.

**Comment by AMDphreak** [comment]
> > Without having read everyone else's comments (only a bit here and there), there is a way to insert new pages. It's strange and unintuitive.
> > ![image](https://private-user-images.githubusercontent.com/1058978/368809938-62be35f0-a4af-45de-84fa-d0b9109a06f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMDc4MTQsIm5iZiI6MTcyNzEwNzUxNCwicGF0aCI6Ii8xMDU4OTc4LzM2ODgwOTkzOC02MmJlMzVmMC1hNGFmLTQ1ZGUtODRmYS1kMGI5MTA5YTA2ZjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjNUMTYwNTE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzcxZDZhNWQ2YTViMzBmZDkzNDYzMDI5YWVhOTgzZGM5OTYzMzQxM2RkNzQyNmU5ZWVjMmUyZWVjNjAwNDU1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.tbZULFgyOHtXCEiY1vgn9ZJFm-Ur6rAdklwX9GZvbCY)
> 
> that would not fix the issue. eg: you're having a very long section of experience. add new page has it's own purpose for making a new section.
> 
> I hope this 'automatic' page break issue will be solved soon, because, the paper size selection would be meaningless..

Yeah I saw the problem. I agree with you. The only workaround right now is to make a custom section called "Work Experience 2" and split your experience up into different sections which requires manually copying the individual job items into a new section, which is dumb. What a pain.

**Comment by PlkMarudny** [comment]
In general, this product seems to be unusable at the moment, cannot get exported multipage PDF, cannot share a public link.

**Comment by ijash-work** [comment]
I think this needs to be prioritized.. 

help me, which particular file needs to be checked, give me a kick start to start contributing.

**Comment by rbenites** [comment]
The work arounds are great and the Public link view works pretty good BUT the best solution would be to be able to export or print the doc the way it appears on the site.
Real world example: I'm trying to print my resume out for a job interview i will have in the next couple days BUT I am spending a lot of time trying to format it so that it prints correctly. Unfortunately IF I export this and share with the people doing the interviews this will not look good for me, or anyone that goes through the same. The workaround for printing works fairly good expect there is an extra page in the file that i cant get rid of unless i use a paid version of Acrobat to modify the PDF.

Note: I am a manager and also do hiring and resume reviews for my team and i would probably skip on a interview if their resume looked broken like this.


Aside from this issue this site/app is GREAT!!!! And I have shared it with a number of people that were recently laid off at the company I work for.

**Comment by cleves0315** [comment]
I tried enabling the sharing feature, then used the browser's print function on the linked webpage. This way, the downloaded PDF will have automatic pagination.

**Comment by radiatewishbone** [comment]
> I tried enabling the sharing feature, then used the browser's print function on the linked webpage. This way, the downloaded PDF will have automatic pagination.

But surely this creates imperfect pages which are cut off at points that cut off information, and potentially doesn't look very well formatted?

Could you please share a link to a demo PDF showing successful pagination using this method? I would be surprised if it actually looked like good formatting.

**Comment by cleves0315** [comment]
> > I tried enabling the sharing feature, then used the browser's print function on the linked webpage. This way, the downloaded PDF will have automatic pagination.
> 
> But surely this creates imperfect pages which are cut off at points that cut off information, and potentially doesn't look very well formatted?
> 
> Could you please share a link to a demo PDF showing successful pagination using this method? I would be surprised if it actually looked like good formatting.

The fact is, as you mentioned, this approach inevitably results in imperfect pagination. For precise control over content layout, consider using the "Toggle Break Line" solution.

**Comment by robertkjr3d** [comment]
Yes, I having the same problem... one long PDF, makes no sense Added a second page, but it was blank.  I might be able try a PDF editor, but the system should just do that for me.  

Edit: Makes it completely useless to share as a service, because you wouldn't want employers to try to print, because only part of the page would come out.  But I have found out how to split with my own PDF editor, and that works and I can I just use the other valuable functions the service does provide.

**Comment by amruthpillai** [comment]
Closing this issue as I'm doing some clean up to move to a newer version of the project. I apologize in advance for the inconvenience if you find the need to reopen this issue.

Please read #2499 for more information about the migration.

**Comment by radiatewishbone** [comment]
@AmruthPillai , when you say you are moving "to a newer version of the project", does that mean you will be including a fix that solves this issue? This is the number one issue reported by users based on the various issue threads, and I have deemed this application unusable for my purposes until this feature is properly integrated. I really hope you are planning to make this improvement in the new version.

**Comment by amruthpillai** [comment]
> [@AmruthPillai](https://github.com/AmruthPillai) , when you say you are moving "to a newer version of the project", does that mean you will be including a fix that solves this issue? This is the number one issue reported by users based on the various issue threads, and I have deemed this application unusable for my purposes until this feature is properly integrated. I really hope you are planning to make this improvement in the new version.

Yes, this should be resolved in the new version for sure.

**Comment by atsaloli** [comment]
Amazing, thanks, @AmruthPillai !

You are helping so many people with Reactive Resume.

I greatly appreciate it!

**Comment by xzessmedia** [comment]
Meanwhile you can split your giant pdf with this https://github.com/pdfarranger/pdfarranger/releases

The problem with reactive resume is that its exporting through a huge canvas instead of using real pages

Thank you very much for your work and effort Amruth!
